src/crimson/net/SocketConnection.h

```diff
@@ -148,52 +158,69 @@ class SocketConnection : public Connection {
   seastar::future<> fault();
 
  public:
-  SocketConnection(Messenger *messenger,
+  SocketConnection(SocketMessenger& messenger,
                    const entity_addr_t& my_addr,
                    const entity_addr_t& peer_addr,
                    seastar::connected_socket&& socket);
   ~SocketConnection();
 
-  bool is_connected() override;
-
-  seastar::future<> client_handshake(entity_type_t peer_type,
-				     entity_type_t host_type) override;
+  Messenger* get_messenger() const override;
 
-  seastar::future<> server_handshake() override;
+  int get_peer_type() const override {
+    return h.connect.host_type;
+  }
 
-  seastar::future<MessageRef> read_message() override;
+  bool is_connected() override;
 
   seastar::future<> send(MessageRef msg) override;
 
   seastar::future<> keepalive() override;
 
   seastar::future<> close() override;
 
-  uint32_t connect_seq() const override {
+ public:
+  /// complete a handshake from the client's perspective
+  seastar::future<> client_handshake(entity_type_t peer_type,
+				     entity_type_t host_type);
+
+  /// complete a handshake from the server's perspective
+  seastar::future<> server_handshake();
+
+  /// read a message from a connection that has completed its handshake
+  seastar::future<MessageRef> read_message();
+
+  /// the number of connections initiated in this session, increment when a
+  /// new connection is established
+  uint32_t connect_seq() const {
     return h.connect_seq;
   }
-  uint32_t peer_global_seq() const override {
+
+  /// the client side should connect us with gseq. it will be reset with a
```

Comment:
please remove " a". see https://github.com/ceph/ceph/pull/24576#discussion_r225104201 , probably you missed it.

Reply:
that was a mistake.. removed a wrong "a"